library/CMakeLists.txt

```diff
@@ -193,10 +193,19 @@ endif()
 if(BUILD_SHARED_LIBS)
   install(TARGETS libf3d
     EXPORT f3dTargets
-    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT library
+    RUNTIME_DEPENDENCY_SET libf3dDeps
+    ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT sdk EXCLUDE_FROM_ALL
     RUNTIME DESTINATION ${CMAKE_INSTALL_BINDIR} COMPONENT library
     LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR} COMPONENT library)
 
+  # Install dependencies, but exclude system libraries
+  set(F3D_DEPENDENCIES_DIR "" CACHE STRING "Semicolon separated directories for dependencies look up")
+  install(RUNTIME_DEPENDENCY_SET libf3dDeps
+    COMPONENT dependencies EXCLUDE_FROM_ALL
+    PRE_EXCLUDE_REGEXES "api-ms-" "ext-ms-"
+    POST_EXCLUDE_REGEXES ".*system32/.*" "^/usr/lib.*" "^/lib.*" "^/var/lib.*"
+    DIRECTORIES "${F3D_DEPENDENCIES_DIR}")
```

Comment:
I'm a bit confused, what files are installed here ?

Reply:
From [CMake](https://cmake.org/cmake/help/latest/command/install.html#runtime-dependency-set):

> The dependencies of targets belonging to a runtime dependency set are installed in the RUNTIME destination and component on DLL platforms, and in the LIBRARY destination and component on non-DLL platforms.

So basically, `objdump`/`dumpbin`/`otool` is run internally to determine dependencies of `libf3d` and copies all of them in `<install>/bin` (Windows) or `<install>/lib` (Linux)  
System libraries are excluded by the regexp.

It's useful when building Python wheels for example where dependencies need to be packaged in the install directory so they are included in the wheel.

Reply:
I had no idea CMake could do that. I suppose it does not fix the rpath and such making it not relocatable.

In any case, lets thouroughly document this component, I dont want people to use that for anything else but its intended purpose.